src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -228,6 +228,31 @@ def update(
         ORMRun.get(name).update(display_name=display_name, description=description, tags=tags, **kwargs)
         return self.get(name)
 
+    @monitor_operation(activity_name="pf.runs.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        runs: Union[str, Run, List[str], List[Run]],
+        **kwargs,
+    ) -> None:
+        """Delete run permanently.
+
+        :param names: run names to delete
+        :param kwargs: other fields to update, fields not supported will be directly dropped.
+        :return: None
+        """
+        if not isinstance(runs, list):
+            runs = [runs]
+        valid_runs = []
+        for run in runs:
+            run = Run._validate_and_return_run_name(run)
+            valid_runs.append((run, LocalStorageOperations(run=self.get(run))))
```

Comment:
Actually, I don't think `LocalStorageOperations `is needed (generating a path and then deleting it, to make matters worse, perhaps just deleting `run. _output_path`), nor is the following `for run, valid_run in valid_runs:` necessary

Just delete the run directly here(`try: run = self.get(run_name), ORMRun.delete(run) catch: pass)`.

![image](https://github.com/microsoft/promptflow/assets/23182548/f982e6dc-fa7c-4691-be4d-e13384f43af5)

Reply:
No... I've guarantee that this run is valid before it comes here.
It will fail in previous step and cannot get in this method.

Reply:
Good catch.